lib/posix/options/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ set(GEN_DIR ${ZEPHYR_BINARY_DIR}/include/generated)
 zephyr_syscall_header_ifdef(CONFIG_POSIX_CLOCK_SELECTION posix_clock.h)
 zephyr_syscall_header_ifdef(CONFIG_POSIX_TIMERS posix_clock.h)
 zephyr_syscall_header_ifdef(CONFIG_XSI_SINGLE_PROCESS posix_clock.h)
+zephyr_syscall_header_ifdef(CONFIG_POSIX_SEMAPHORES posix_clock.h)
```

Comment:
Given that `POSIX_TIMERS` and `POSIX_SEMAPHORES` are independent Option Groups, I think this line needs to be removed.

https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_subprofiles.html

Reply:
yes, sorry, I actually thought I'd removed this with the update, but I failed to push the change :-/